lumbermill-aws/src/main/java/lumbermill/internal/aws/S3ClientImpl.java

```diff
@@ -113,13 +121,9 @@ public void init() {
                         LOGGER.debug("Deleted s3 file s3://{}/{} successfully", sBucket, sKey);
                     }
                 })
-                .doOnUnsubscribe(() -> {
+                .doAfterTerminate(() -> {
```

Comment:
If you un-comment the `.observeOn(Schedulers.io())` in s3poll.groovy then you can see that the downloaded (local, temporary) file is removed before the end of the pipeline.

```
2017-03-10 19:46:35.908 [lumbermill-s3poll-0] TRACE l.internal.aws.S3ScheduledPoll - Polling files under s3://s3poll-bucket/files/ available permits: 2
2017-03-10 19:46:44.409 [lumbermill-s3poll-0] TRACE lumbermill.internal.aws.S3ClientImpl - Getting file from s3://s3poll-bucket/files/test.md
2017-03-10 19:46:44.686 [lumbermill-s3poll-0] DEBUG lumbermill.internal.aws.S3ClientImpl - File download from s3 was stored in /tmp/lumbermill1428435659593556344.log
{
  "bucketName" : "s3poll-bucket",
  "key" : "files/test.md",
  "size" : 1536,
  "lastModified" : 1489171595000,
  "storageClass" : "STANDARD",
  "owner" : {
    "displayName" : "...",
    "id" : "..."
  },
  "etag" : "8034c79140e62bd590ce79497a3b907b"
}
2017-03-10 19:46:44.696 [lumbermill-s3poll-0] DEBUG lumbermill.internal.aws.S3ClientImpl - Uploading to s3://s3poll-bucket/processed/files/test.md
2017-03-10 19:46:44.839 [lumbermill-s3poll-0] DEBUG lumbermill.internal.aws.S3ClientImpl - Deleting object s3://s3poll-bucket/files/test.md
2017-03-10 19:46:44.981 [lumbermill-s3poll-0] DEBUG lumbermill.internal.aws.S3ClientImpl - Deleted s3 file s3://s3poll-bucket/files/test.md successfully
2017-03-10 19:46:44.981 [lumbermill-s3poll-0] DEBUG lumbermill.internal.aws.S3ClientImpl - Unsubscribed:Deleted local file /tmp/lumbermill1428435659593556344.log successfully ? true
done sleeping
2017-03-10 19:46:52.856 [RxIoScheduler-2] TRACE l.internal.aws.S3ScheduledPoll - release()
```